### PR TITLE
yargs: Add the ArgvFromBuilder type that is useful when using command modules

### PR DESCRIPTION
**Pull request by segevfiner**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/yargs/yargs/blob/main/docs/advanced.md#commanddirdirectory-opts
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

This can be used like this:
```ts
import type { ArgumentsCamelCase, Argv, ArgvFromBuilder } from 'yargs';

export const command = 'foo';

export const desc = 'Does foo!';

export function builder(args: Argv) {
  return args.option('bar', { type: 'number' });
}

export function handler(argv: ArgvFromBuilder<typeof builder>) {
  console.log('bar:', argv.bar); // <-- argv.bar will be properly typed, at least as much as when declared directly using `command('foo', 'Does foo!', (yargs) => { ... }, (argv) => { ... })`
}
```

Do let me know if I defined that type correctly, or if its name is alright. TypeScript type manipulations like this are always quite a handful...

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/yargs/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9e87b322b94b3bbd8723a052e4b9bc909815d853
</p>

**Comment by forivall** [comment]
IMO, this should be left inside of your code instead of bringing it into the `yargs` typings. Otherwise, the type itself looks good, but I'd go with the name `BuilderArguments` or `BuilderArgumentsType` as those names are closer to the typescript builtin names of `Parameters` and `ReturnType`. Also `Argv` is specifically for the `yargs.Argv` api object, but this type is extracting the `Arguments` type.

**Comment by segevfiner** [comment]
Because its kinda complex, I'd rather it will be available for others and the best place seems to be here... I'll rename later

**Review by forivall — CHANGES_REQUESTED** [review]
Needs tests

File: types/yargs/yargs-tests.ts [hunk 1]
```diff
@@ -1618,3 +1618,13 @@ function Argv$commandCommandModuleArray() {
...
     yargs.command([CommandOne, CommandTwo]);
 }
+
+function BuilderArguments() {
+    const builder = (yargs: yargs.Argv) => {
+        return yargs.option("a", { default: "foo" });
+    };
+
+    const handler = (argv: BuilderArguments<typeof builder>) => {
+        argv.a;
```
Comment (forivall) [inline types/yargs/yargs-tests.ts:1628]: Could have an `$ExpectType` assertion here.

**Review by forivall — APPROVED** [review]

**Review by forivall — APPROVED** [review]

**Review by sheetalkamat — APPROVED** [review]

**Comment by segevfiner** [comment]
Ready to merge

**Pull request merged**